module/programs/helper.js

```diff
@@ -341,6 +344,54 @@ module.exports = class ProgramsHelper {
 					}
 				}
 
+				if (data.components) {
+					let programDocumentRecord = await programsQueries.programsDocument({ _id: programId }, [
+						'components',
+					])
+
+					if (!programDocumentRecord[0]) {
+						throw {
+							message: CONSTANTS.apiResponses.PROGRAM_NOT_UPDATED,
+						}
+					}
+
+					let currentComponents = [...programDocumentRecord[0].components] || [] // Create a copy
+
+					// Process each component in data.components
+					for (let component of data.components) {
+						let componentIndex = currentComponents.findIndex(
+							(c) => c._id.toString() === component._id.toString()
```

Comment:
@borkarsaish65 here also you are using variable name as "c" please avoid this. this is not standard

Reply:
done